internal/service/rds/custom_db_engine_version.go

```diff
@@ -0,0 +1,469 @@
+		}
+		input.Manifest = aws.String(file)
+	} else if v, ok := d.GetOk("manifest"); ok {
+		input.Manifest = aws.String(v.(string))
+	}
+
+	output, err := conn.CreateCustomDBEngineVersionWithContext(ctx, &input)
+	if err != nil {
+		return append(diags, create.DiagError(names.RDS, create.ErrActionCreating, ResNameCustomDBEngineVersion, fmt.Sprintf("%s:%s", aws.StringValue(output.Engine), aws.StringValue(output.EngineVersion)), err)...)
+	}
+
+	if output == nil {
+		return append(diags, create.DiagError(names.RDS, create.ErrActionCreating, ResNameCustomDBEngineVersion, fmt.Sprintf("%s:%s", aws.StringValue(output.Engine), aws.StringValue(output.EngineVersion)), errors.New("empty output"))...)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(output.Engine), aws.StringValue(output.EngineVersion)))
+
+	if _, err := waitCustomDBEngineVersionCreated(ctx, conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return append(diags, create.DiagError(names.RDS, create.ErrActionWaitingForCreation, ResNameCustomDBEngineVersion, d.Id(), err)...)
+	}
+
+	return append(diags, resourceCustomDBEngineVersionRead(ctx, d, meta)...)
+}
+
+func resourceCustomDBEngineVersionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).RDSConn(ctx)
+
+	out, err := FindCustomDBEngineVersionByID(ctx, conn, d.Id())
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] RDS CustomDBEngineVersion (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return diags
+	}
+
+	if err != nil {
+		return append(diags, create.DiagError(names.RDS, create.ErrActionReading, ResNameCustomDBEngineVersion, d.Id(), err)...)
+	}
+
+	d.Set("arn", out.DBEngineVersionArn)
+	if out.CreateTime != nil {
+		d.Set("create_time", out.CreateTime.Format(time.RFC3339))
+	}
+	d.Set("database_installation_files_s3_bucket_name", out.DatabaseInstallationFilesS3BucketName)
+	d.Set("database_installation_files_s3_prefix", out.DatabaseInstallationFilesS3Prefix)
+	d.Set("db_parameter_group_family", out.DBParameterGroupFamily)
+	d.Set("description", out.DBEngineVersionDescription)
+	d.Set("engine", out.Engine)
+	d.Set("engine_version", out.EngineVersion)
+	d.Set("image_id", out.Image.ImageId)
+	d.Set("kms_key_id", out.KMSKeyId)
+	d.Set("major_engine_version", out.MajorEngineVersion)
+	d.Set("manifest_computed", out.CustomDBEngineVersionManifest)
+	d.Set("status", out.Status)
+
+	setTagsOut(ctx, out.TagList)
+
+	return diags
+}
+
+func resourceCustomDBEngineVersionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).RDSConn(ctx)
+
+	if d.HasChangesExcept("description", "status") {
+		return append(diags, create.DiagError(names.RDS, create.ErrActionUpdating, ResNameCustomDBEngineVersion, d.Id(), errors.New("only description and status can be updated"))...)
+	}
```

Comment:
CEVs are immutable artifacts (AMIs) that if deleted with automation will cause downstream impacts to resources (instances). Once created, only the Description and Status can be modified. The design decision was made to not use ForceNew as it would be counter intuitive for the users. The primary user here is a DBA with little to no Cloud Experience, and maybe zero Terraform/IaC experience.